File: GestionStore/src/main/java/entities/detailfacture.java
package entities;

public class detailfacture {

    private int idFacture;
    private int idDetailFacture;
    private int idProduit;
    private float prixVenteUnitaire;
    private int quantite;
    private float tauxRemise;
    private float prixtotalArticle;

    public int getIdFacture() {
        return idFacture;
    }

    public void setIdFacture(int idFacture) {
        this.idFacture = idFacture;
    }

    public int getIdDetailFacture() {
        return idDetailFacture;
    }

    public void setIdDetailFacture(int idDetailFacture) {
        this.idDetailFacture = idDetailFacture;
    }

    public int getIdProduit() {
        return idProduit;
    }

    public void setIdProduit(int idProduit) {
        this.idProduit = idProduit;
    }

    public float getPrixVenteUnitaire() {
        return prixVenteUnitaire;
    }

    public void setPrixVenteUnitaire(float prixVenteUnitaire) {
        this.prixVenteUnitaire = prixVenteUnitaire;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public float getTauxRemise() {
        return tauxRemise;
    }

    public void setTauxRemise(float tauxRemise) {
        this.tauxRemise = tauxRemise;
    }

    public float getPrixtotalArticle() {
        return prixtotalArticle;
    }

    public void setPrixtotalArticle(float prixtotalArticle) {
        this.prixtotalArticle = prixtotalArticle;
    }

    public detailfacture() {
    }

    public detailfacture(int idFacture, int idDetailFacture, int idProduit, float prixVenteUnitaire, int quantite, float tauxRemise, float prixtotalArticle) {
        this.idFacture = idFacture;
        this.idDetailFacture = idDetailFacture;
        this.idProduit = idProduit;
        this.prixVenteUnitaire = prixVenteUnitaire;
        this.quantite = quantite;
        this.tauxRemise = tauxRemise;
        this.prixtotalArticle = prixtotalArticle;
    }

    public detailfacture(int idFacture, int idDetailFacture, int idProduit, int quantite, float tauxRemise) {
        this.idFacture = idFacture;
        this.idDetailFacture = idDetailFacture;
        this.idProduit = idProduit;
        this.quantite = quantite;
        this.tauxRemise = tauxRemise;
    }

    @Override
    public String toString() {
        return "detailfacture{" +
                "idFacture=" + idFacture +
                ", idDetailFacture=" + idDetailFacture +
                ", idProduit=" + idProduit +
                ", prixVenteUnitaire=" + prixVenteUnitaire +
                ", quantite=" + quantite +
                ", tauxRemise=" + tauxRemise +
                ", prixtotalArticle=" + prixtotalArticle +
                '}';
    }
}

File: webapp/src/GymPlus/src/Entity/Detailfacture.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\DetailfactureRepository;


#[ORM\Entity(repositoryClass: DetailfactureRepository::class)]
class Detailfacture
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $iddetailfacture;

    #[ORM\Column]
    private ?float $prixventeunitaire;

    #[ORM\Column]
    private ?int $quantite;

    #[ORM\Column]
    private ?float $tauxremise;

    #[ORM\Column]
    private ?float $prixtotalarticle;

    #[ORM\ManyToOne(targetEntity: Facture::class)]
    private $idfacture;

    #[ORM\ManyToOne(targetEntity: Produit::class)]
    private $idproduit;

    public function getIddetailfacture(): ?int
    {
        return $this->iddetailfacture;
    }

    public function getPrixventeunitaire(): ?float
    {
        return $this->prixventeunitaire;
    }

    public function setPrixventeunitaire(?float $prixventeunitaire): static
    {
        $this->prixventeunitaire = $prixventeunitaire;

        return $this;
    }

    public function getQuantite(): ?int
    {
        return $this->quantite;
    }

    public function setQuantite(?int $quantite): static
    {
        $this->quantite = $quantite;

        return $this;
    }

    public function getTauxremise(): ?float
    {
        return $this->tauxremise;
    }

    public function setTauxremise(?float $tauxremise): static
    {
        $this->tauxremise = $tauxremise;

        return $this;
    }

    public function getPrixtotalarticle(): ?float
    {
        return $this->prixtotalarticle;
    }

    public function setPrixtotalarticle(?float $prixtotalarticle): static
    {
        $this->prixtotalarticle = $prixtotalarticle;

        return $this;
    }

    public function getIdfacture(): ?Facture
    {
        return $this->idfacture;
    }

    public function setIdfacture(?Facture $idfacture): static
    {
        $this->idfacture = $idfacture;

        return $this;
    }

    public function getIdproduit(): ?Produit
    {
        return $this->idproduit;
    }

    public function setIdproduit(?Produit $idproduit): static
    {
        $this->idproduit = $idproduit;

        return $this;
    }


}
